lib/svg-sprite/queue.js

```diff
@@ -19,88 +19,76 @@ const Shape = require('./shape.js');
  *
  * @param {SVGSpriter} spriter          SVGSpriter instance
  */
-function SVGSpriterQueue(spriter) {
-    this._spriter = spriter;
-    this._files = [];
-    this.active = 0;
+module.exports = class SVGSpriterQueue extends EventEmitter {
+    constructor(spriter) {
+        super();
 
-    events.EventEmitter.call(this);
-    this.on('add', this.process.bind(this));
-    this.on('remove', this.process.bind(this));
+        this._spriter = spriter;
+        this._files = [];
+        this.active = 0;
 
-    this._spriter.debug('Created processing queue instance');
-}
+        this.on('add', this.process.bind(this));
+        this.on('remove', this.process.bind(this));
 
-/**
- * Prototype
- *
- * @type {object}
- */
-SVGSpriterQueue.prototype = Object.create(events.EventEmitter.prototype);
+        this._spriter.debug('Created processing queue instance');
+    }
 
-/**
- * Add a shape to the processing queue
- *
- * @param {File} file                   Shape file
- */
-SVGSpriterQueue.prototype.add = function(file) {
-    this._spriter.debug('Added "%s" to processing queue', path.basename(file.path));
-    this._files.push(file);
-    this.emit('add');
-};
+    /**
+     * Add a shape to the processing queue
+     *
+     * @param {File} file                   Shape file
+     */
+    add(file) {
+        this._spriter.debug('Added "%s" to processing queue', path.basename(file.path));
+        this._files.push(file);
+        this.emit('add');
+    }
 
-/**
- * Try to process an item in the queue
- */
-SVGSpriterQueue.prototype.process = function() {
-    if (this._files.length && this.active < this._spriter._limit) {
-        ++this.active;
-        const file = this._files.shift();
-        let shape;
-        let spriter;
+    /**
+     * Try to process an item in the queue
+     */
+    process() {
+        if (this._files.length && this.active < this._spriter._limit) {
```

Comment:
Maybe it makes sense to return early here? Feel free to push any patches to this branch.